### PR TITLE
Add hosted mode

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -55,7 +55,19 @@ type ClusterManagerSpec struct {
...
+	// RegistrationWebhookConfiguration represents the customized webhook-server configuration of registration.
+	// +optional
+	RegistrationWebhookConfiguration WebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
+
+	// WorkWebhookConfiguration represents the customized webhook-server configuration of work.
+	// +optional
+	WorkWebhookConfiguration WebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
+}
+
+// DetachedClusterManagerConfiguration represents customized configurations we need to set for clustermanager in the Hosted mode.
```
Comment (qiujian16) [inline operator/v1/types.go]: I think it is save to remove this part. I do not think we have another consumer in old release that uses this field

**Review by qiujian16 — COMMENTED** [review]
please add some integration test for this.

File: operator/v1/types.go (hunk 1 above)
Comment (xuezhaojun) [reply operator/v1/types.go]: Sure. let me remove it.

File: operator/v1/types.go [hunk 2]
```diff
@@ -126,7 +126,12 @@ const (
 
 	// InstallModeDetached means deploying components outside.
 	// The cluster-manager will be deployed outside of the hub-cluster, the klusterlet will be deployed outside of the managed-cluster.
+	// DEPRECATED: please use Hosted instead.
 	InstallModeDetached InstallMode = "Detached"
+
+	// InstallModeDetached means deploying components outside.
```
Comment (zhujian7) [inline operator/v1/types.go]: InstallModeHosted

Comment (xuezhaojun) [reply operator/v1/types.go]: Yes. fixed.

File: operator/v1/types.go [hunk 3]
```diff
@@ -98,22 +98,22 @@ type KlusterletDeployOption struct {
...
+	// In Hosted mode, only crd and configurations are installed on one cluster(defined as hub-cluster). Controllers run in another
 	// cluster (defined as management-cluster) and connect to the hub with the kubeconfig in secret of "external-hub-kubeconfig"(a kubeconfig
 	// of hub-cluster with cluster-admin permission).
 	// Note: Do not modify the Mode field once it's applied.
 	// +required
 	// +default=Default
 	// +kubebuilder:validation:Required
 	// +kubebuilder:default=Default
-	// +kubebuilder:validation:Enum=Default;Detached
+	// +kubebuilder:validation:Enum=Default;Detached;Hosted
```
Comment (zhujian7) [inline operator/v1/types.go]: Do we still need the `Detached` value for cluster manager?

Comment (xuezhaojun) [reply operator/v1/types.go]: That's a good point! Removed Detached mode.

**Review by qiujian16 — COMMENTED** [review]
looks good, please also add a klusterlet_test

**Pull request merged**